src/Api/InitializeInTracee.cpp

```diff
@@ -4,16 +4,47 @@
 
 #include "Api/InitializeInTracee.h"
 
+#include <functional>
+
 #include "Api/Orbit.h"
-#include "OrbitBase/Result.h"
+#include "OrbitBase/ExecutablePath.h"
+#include "OrbitBase/UniqueResource.h"
+#include "UserSpaceInstrumentation/Attach.h"
+#include "UserSpaceInstrumentation/ExecuteInProcess.h"
+#include "UserSpaceInstrumentation/InjectLibraryInTracee.h"
 
+using namespace orbit_user_space_instrumentation;
+using orbit_grpc_protos::ApiTableInfo;
 using orbit_grpc_protos::CaptureOptions;
 
 namespace orbit_api {
+ErrorMessageOr<void> InitializeInTracee(const CaptureOptions& capture_options) {
+  if (capture_options.api_table_infos().size() == 0) {
+    return ErrorMessage("No api table to initilize.");
+  }
+
+  ErrorMessageOr<void> result = outcome::success();
+  int32_t pid = capture_options.pid();
+
+  OUTCOME_TRY(AttachAndStopProcess(pid));
+  {
+    // Make sure we resume the target process, even on early-outs.
+    orbit_base::unique_resource scope_exit{
+        pid, [&result](int32_t pid) { result = DetachAndContinueProcess(pid); }};
```

Comment:
I feel like we want a proper scope_guard here (one that does not need a resource)

Reply:
Do we have one?

Reply:
not yet :)

Reply:
Once we have it I'll be happy to use it :-)